src/rdkafka_buf.c

```diff
@@ -407,9 +408,12 @@ void rd_kafka_buf_handle_op (rd_kafka_op_t *rko, rd_kafka_resp_err_t err) {
         response = request->rkbuf_response; /* May be NULL */
         request->rkbuf_response = NULL;
 
-        rd_kafka_buf_callback(request->rkbuf_rkb->rkb_rk,
-			      request->rkbuf_rkb, err,
-                              response, request);
+        if (!(rk = rko->rko_rk)) {
```

Comment:
haven't looked through any of the details, but the need for this kind of logic seems unfortunate.

Reply:
Yeah, we should probably always set rko_rk on ops, but that's a bigger change for later.